api/src/apps/routes.rs

```diff
@@ -513,4 +513,522 @@ mod tests {
+                forwarded_proto, forwarded_host, forwarded_port
+            );
+
+            let expected_url = format!(
+                "{}{}/{}/",
+                expected_url1,
+                "master".to_string(),
+                "service-a".to_string()
+            );
+            let expected_response = json!({
+                "master": [
+                    {
+                        "url": expected_url
+                    }
+                ]
+            });
+
+            assert_json_include!(actual: json, expected: expected_response);
+
+            Ok(())
+        }
+
+        #[tokio::test]
+        async fn host_header_response_with_all_default_values(
+        ) -> Result<(), crate::apps::AppsServiceError> {
+            let (host_meta_cache, mut host_meta_crawler) = crate::host_meta_crawling();
+            let infrastructure = Box::new(Dummy::new());
+            let apps = Arc::new(AppsService::new(Default::default(), infrastructure).unwrap());
+            let _result = apps
+                .create_or_update(
+                    &AppName::from_str("master").unwrap(),
+                    &AppStatusChangeId::new(),
+                    None,
+                    &vec![sc!("service-a")],
+                )
+                .await?;
+            host_meta_crawler.fake_host_meta_info("master".to_string(), "service-a".to_string());
+            let rocket = rocket::build()
+                .manage(host_meta_cache)
+                .manage(apps)
+                .mount("/", routes![crate::apps::routes::apps])
+                .mount("/api/apps", crate::apps::apps_routes());
+
+            let client = Client::tracked(rocket).await.expect("valid rocket");
+
+            let get = client
+                .get("/")
+                .header(rocket::http::Header::new("host", "localhost"))
+                .header(ContentType::JSON)
+                .dispatch();
+
+            let response = get.await;
+
+            let body_str = response.into_string().await.expect("valid response body");
+            let my_json: Result<Value, _> = serde_json::from_str(&body_str);
+            assert!(my_json.is_ok(), "Failed to parse JSON response");
+            let json = my_json.unwrap();
+            let forwarded_proto = "http";
+            let forwarded_host = "localhost";
+            let expected_url1 = format!("{}://{}/", forwarded_proto, forwarded_host,);
+
+            let expected_url = format!(
+                "{}{}/{}/",
+                expected_url1,
+                "master".to_string(),
+                "service-a".to_string()
+            );
+            let expected_response = json!({
+                "master": [
+                    {
+                        "url": expected_url
+                    }
+                ]
+            });
+
+            assert_json_include!(actual: json, expected: expected_response);
+
+            Ok(())
+        }
+
+        #[tokio::test]
+        async fn without_prefer_header() {
```

Comment:
Typo or copy paste error?

```suggestion
        async fn bad_request_without_host_header() {
```